File: frameworks/domain.core/src/main/java/com/example/frameworks/domain/core/IntFieldType.java
package com.example.frameworks.domain.core;

public class IntFieldType extends FieldType<Integer> {
    public static IntFieldType asInt() {
        return new IntFieldType();
    }

    @Override
    protected boolean match(Class<?> underlyingType) {
        return Number.class.isAssignableFrom(underlyingType);
    }

    @Override
    protected Integer convert(Object value) {
        return ((Number) value).intValue();
    }
}

File: frameworks/payload.core/src/main/kotlin/com/qomolangma/frameworks/payload/core/IntFieldType.kt
package com.qomolangma.frameworks.payload.core

class IntFieldType : FieldType<Int>() {
    override fun match(underlyingType: Class<*>): Boolean {
        return Number::class.java.isAssignableFrom(underlyingType)
    }

    override fun convert(value: Any): Int {
        return (value as Number).toInt()
    }

    companion object {
        @JvmStatic
        fun asInt(): IntFieldType {
            return IntFieldType()
        }
    }
}
